rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -680,11 +675,13 @@ void store_security_filepath_in_qos(
     allocator.deallocate(security_file_path, allocator.state);
   }
 }
+#endif  /* DDS_HAS_SECURITY */
 
 /*  Set all the qos properties needed to enable DDS security  */
-void configure_qos_for_security(
+rmw_ret_t configure_qos_for_security(
   dds_qos_t * qos, const rmw_node_security_options_t * security_options)
 {
+#if DDS_HAS_SECURITY
```

Comment:
He wants us to use `DDS_HAS_PROPERTY_LIST_QOS` here.

```suggestion
#if DDS_HAS_SECURITY && DDS_HAS_PROPERTY_LIST_QOS
```